example/heatEquation/src/heatEquation.cpp

```diff
@@ -41,19 +41,18 @@
 
 struct HeatEquationKernel
 {
-    template<typename TAcc>
+    template<typename TAcc, typename TAccessorCurr, typename TAccessorNext>
     ALPAKA_FN_ACC auto operator()(
         TAcc const& acc,
-        double const* const uCurrBuf,
-        double* const uNextBuf,
-        uint32_t const extent,
+        TAccessorCurr uCurrBuf,
+        TAccessorNext uNextBuf,
         double const dx,
         double const dt) const -> void
```

Comment:
I am starting to get scared. We loose a lot of type information by templating the accessor. We do not know if the access is `const`, and we do not even now the element type.